contracts/Strategy.sol

```diff
@@ -561,183 +451,136 @@ contract Strategy is BaseStrategy, IERC3156FlashBorrower {
-                _leverUpStep(totalAmountToBorrow)
-            );
-
-            if (totalAmountToBorrow > minWant) {
-                totalAmountToBorrow = totalAmountToBorrow.sub(
-                    _leverUpFlashLoan(totalAmountToBorrow)
-                );
-            }
-        } else {
-            for (
-                uint8 i = 0;
-                i < maxIterations && totalAmountToBorrow > minWant;
-                i++
-            ) {
-                totalAmountToBorrow = totalAmountToBorrow.sub(
-                    _leverUpStep(totalAmountToBorrow)
-                );
-            }
-        }
-    }
-
-    function _leverUpFlashLoan(uint256 amount) internal returns (uint256) {
-        (uint256 deposits, uint256 borrows) = getCurrentPosition();
-        uint256 depositsToMeetLtv =
-            getDepositFromBorrow(borrows, maxBorrowCollatRatio);
-        uint256 depositsDeficitToMeetLtv = 0;
-        if (depositsToMeetLtv > deposits) {
-            depositsDeficitToMeetLtv = depositsToMeetLtv.sub(deposits);
-        }
-        return
-            FlashMintLib.doFlashMint(
-                false,
-                amount,
-                address(want),
-                daiBorrowCollatRatio,
-                depositsDeficitToMeetLtv
-            );
-    }
-
-    function _leverUpStep(uint256 amount) internal returns (uint256) {
-        if (amount == 0) {
-            return 0;
-        }
-
-        uint256 wantBalance = balanceOfWant();
+        for (
+            uint8 i = 0;
+            i < maxIterations && totalAmountToBorrow > minWant;
+            i++
+        ) {
+            uint256 amount = totalAmountToBorrow;
 
-        // calculate how much borrow can I take
-        (uint256 deposits, uint256 borrows) = getCurrentPosition();
-        uint256 canBorrow =
-            getBorrowFromDeposit(
+            // calculate how much borrow to take
+            //(deposits, borrows) = getCurrentPosition();
+            uint256 canBorrow = getBorrowFromDeposit(
                 deposits.add(wantBalance),
                 maxBorrowCollatRatio
             );
 
-        if (canBorrow <= borrows) {
-            return 0;
-        }
-        canBorrow = canBorrow.sub(borrows);
+            if (canBorrow <= borrows) {
+                break;
+            }
+            canBorrow = canBorrow.sub(borrows);
 
-        if (canBorrow < amount) {
-            amount = canBorrow;
-        }
+            if (canBorrow < amount) {
+                amount = canBorrow;
+            }
 
-        // deposit available want as collateral
-        _depositCollateral(wantBalance);
+            // deposit available want as collateral
+            _depositCollateral(wantBalance);
 
-        // borrow available amount
-        _borrowWant(amount);
+            // borrow available amount
+            _borrowWant(amount);
 
-        return amount;
+            // track ourselves to save gas
+            deposits = deposits.add(wantBalance);
```

Comment:
not that useful in FTM but still nice. 

are we adding unnecessary risks here? (rounding errors or similar)

Reply:
That's my concern too. In some sense, the lack of flashloans pushed me to make these optimizations, so we could do extra loop iterations.

Reply:
I think we should be fine. let's leave it like this.